File: 169-majority-element/majority-element.py
class Solution:
    def MaxFreq(self,freq):
        Maxf = 0
        Maxk = None
        for key, value in freq.items():
            if value>Maxf:
                Maxf=value
                Maxk=key
        return Maxk

    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        for num in nums:
            if num not in freq:
                freq[num] = 1
            else :
                freq[num]+=1
        return self.MaxFreq(freq)
        

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        unordered_map<int, int> freq;
        int maxFreq=0;
        int ans=0;
        for(int i=0; i<nums.size(); i++){
            freq[nums[i]]++;
        }
        for(auto it:freq){
            if(it.second>maxFreq){
                maxFreq=it.second;
                ans=it.first;
            }
        }
        return ans;
    }
};